app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save{email.email = email.downcase}
```

Comment:
```suggestion
  before_save{email.downcase!}
```
so sánh giúp a: downcase vs downcase!

Reply:
downcase và downcase! dùng để in thường các ký tự trong chuỗi
Tuy nhiên downcase! có thể thay đổi giá trị của chuỗi ban đầu còn downcase thì không,
downcase! sẽ trả về nil nếu xâu ký tự không có gì thay đổi